src/software/ai/hl/stp/action/intercept_ball_action.cpp

```diff
@@ -35,146 +36,132 @@ void InterceptBallAction::accept(MutableActionVisitor& visitor)
     visitor.visit(*this);
 }
 
-std::optional<Point> InterceptBallAction::getPointBallLeavesField(const Field& field,
-                                                                  const Ball& ball)
+void InterceptBallAction::interceptSlowBall(IntentCoroutine::push_type& yield)
 {
-    Ray ball_ray(ball.position(), ball.velocity());
-    if (field.pointInFieldLines(ball.position()))
+    // Run into the ball until the ball is in the dribbler and the ball's velocity
+    // is similar to the robot's (so we know we have control of the ball)
+    bool intercept_done = false;
+    do
     {
-        std::unordered_set<Point> intersections =
-            intersection(field.fieldLines(), ball_ray);
-        if (!intersections.empty())
+        auto face_ball_orientation = (ball.position() - robot->position()).orientation();
+        yield(std::make_unique<MoveIntent>(
+            robot->id(), ball.position(), face_ball_orientation, 0, 0, DribblerEnable::ON,
+            MoveType::NORMAL, AutokickType::NONE, BallCollisionType::ALLOW));
+
+        if (ball.velocity().length() > BALL_MOVING_SLOW_SPEED_THRESHOLD * 2)
```

Comment:
Quick comment here on what this case represents physically would be helpful.

Reply:
done

Reply:
done